frontend/src/makecolors.js

```diff
@@ -0,0 +1,49 @@
+/* eslint-disable no-console */
+const fs = require('fs');
+const colors = require('./colors');
+
+function getPropName(key) {
+  let propName = '';
+
+  for (let x = 0; x < key.length; x += 1) {
+    const c = key.charAt(x);
+
+    if (c === c.toUpperCase()) {
+      propName += `-${c.toLowerCase()}`;
+    } else {
+      propName += c;
+    }
+  }
+
+  return propName;
+}
+
+function makeColors() {
+  const keys = Object.keys(colors);
+  const values = Object.values(colors);
+
+  const contents = `
+// STOP! Don't edit this file.
+// Instead, edit colors.js and run 'yarn makecolors'
+// from the frontend directory to generate this file'
+
+// figma reference here:
+// https://www.figma.com/file/5Fr0NKQf9MQ5WGd8BWxA6i/TTA_SmartHub-Library-09132021?node-id=0%3A14
+
+${keys.map((key, index) => {
```

Comment:
Can we indent this or will it cause an indention in the resulting file?

Reply:
yeah the ``` `` ``` template strings respect the indentation you pass in

Reply:
Can we remove the indention from the other code (line 34-37) to match line 33? Took me a while to realize the block that starts at line 33 does _not_ end at line 47.